tokio/src/sync/broadcast.rs

```diff
@@ -746,22 +794,49 @@ where
     /// }
     /// ```
     pub fn try_recv(&mut self) -> Result<T, TryRecvError> {
-        let guard = self.recv_ref()?;
+        let guard = self.recv_ref(None)?;
         guard.clone_value().ok_or(TryRecvError::Closed)
     }
 
-    #[doc(hidden)] // TODO: document
+    #[doc(hidden)]
+    #[deprecated(since = "0.2.21", note = "use async fn recv()")]
```

Comment:
It occurs to me that we could also provide a `poll_recv`-type function for `Pin<&mut Self>` as an alternative, and when we remove the deprecated method, replace it with that? Then, we could remove the `Pin<Box<...>>` from around the waiter and just project the waiter field?

Reply:
I'm not following.

Reply:
Sorry if that wasn't clear. Here's what I was thinking: We could add a new `poll`-based method that requires the `Receiver` to be pinned, which would eventually allow us to continue providing a polling interface. When we remove the deprecated `poll_recv` message in 0.3, the polling-based method that requires the receiver to be pinned would not have to box a waiter; it could simply pin project to a waiter field.

Reply:
Ah, I would rather hold off on adding a poll based fn for now and figure out a longer term strategy that we can apply universally across Tokio.

I wonder if we could provide a utility that handles the pinning across the board.

Reply:
Also, I'm not sure how you would add a poll based method that requires the receiver to be pinned. We are specifically avoiding having to pin the receiver. The problem is we have nowhere to store the waiter w/o the future struct.